modules/exploits/windows/http/exchange_ecp_viewstate.rb

```diff
@@ -0,0 +1,364 @@
+    length_prefixed_string :name
+    int32                  :member_count, :value => lambda { member_names.length }
+    array                  :member_names, :type => :length_prefixed_string, :read_until => lambda { index == member_count - 1 }
+  end
+
+  class MemberTypeInfo < BinData::Record
+    endian                 :little
+    # see: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/aa509b5a-620a-4592-a5d8-7e9613e0a03e
+    array                  :binary_type_enums, :type => :uint8
+    #??? : member_type_info # this field is not supported, it's only used if binary_type_enums contains
+    # Primitive, SystemClass, Class, or PrimitiveArray
+    virtual                :valid, :assert => lambda {
+      (!binary_type_enums.include? BinaryTypeEnum[:Primitive]) \
+        && (!binary_type_enums.include? BinaryTypeEnum[:SystemClass]) \
+        && (!binary_type_enums.include? BinaryTypeEnum[:Class]) \
+        && (!binary_type_enums.include? BinaryTypeEnum[:PrimitiveArray])
+    }
+    virtual :assert => lambda { valid.assert! }
+  end
+
+  class ClassWithMembersAndTypes < BinData::Record
+    # see: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/847b0b6a-86af-4203-8ed0-f84345f845b9
+    endian                 :little
+    hide                   :record_type
+    uint8                  :record_type, :asserted_value => RecordTypeEnum[:ClassWithMembersAndTypes]
+    class_info             :class_info
+    member_type_info       :member_type_info
+    int32                  :library_id
+    virtual                :valid, :assert => lambda {
+      member_type_info.valid.assert!
+    }
+    virtual :assert => lambda { valid.assert! }
+  end
+
+  class SerializationHeaderRecord < BinData::Record
+    # see: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/a7e578d3-400a-4249-9424-7529d10d1b3c
+    endian                 :little
+    default_parameter      major_version: 1
+    default_parameter      minor_version: 0
+    hide                   :record_type
+    uint8                  :record_type, :asserted_value => RecordTypeEnum[:SerializedStreamHeader]
+    int32                  :header_id
+    int32                  :major_version, :initial_value => :major_version
+    int32                  :minor_version, :initial_value => :minor_version
+  end
+
+  class ObjectStateFormatter < BinData::Record
+    # see: https://github.com/microsoft/referencesource/blob/3b1eaf5203992df69de44c783a3eda37d3d4cd10/System.Web/UI/ObjectStateFormatter.cs
+    endian                 :little
+    default_parameter      marker_format: 0xff
+    default_parameter      marker_version: 1
+    hide                   :marker_format,  :marker_version
+    uint8                  :marker_format,  :initial_value => :marker_format
+    uint8                  :marker_version, :initial_value => :marker_version
+    uint8                  :token
+  end
+
+  def check
+    return Exploit::CheckCode::Unsupported
```

Comment:
Maybe add a reason like `Exploit::CheckCode::Unsupported('Blah blah')`. You can also drop the `return`.